docs/framework-integrations/nextjs.mdx

```diff
@@ -0,0 +1,424 @@
+
+	const body = await request.json();
+	const params = JSON.stringify({
+		auth: {
+			key: authKey,
+			expires,
+		},
+		template_id: templateId,
+		fields: {
+			// You can use this in your template.
+			userId: body.userId,
+		},
+		// your other params like notify_url, etc.
+	});
+
+	const signatureBytes = crypto
+		.createHmac('sha384', authSecret)
+		.update(Buffer.from(params, 'utf-8'));
+	// The final signature needs the hash name in front, so
+	// the hashing algorithm can be updated in a backwards-compatible
+	// way when old algorithms become insecure.
+	const signature = `sha384:${signatureBytes.digest('hex')}`;
+
+	return NextResponse.json({ expires, signature, params });
+}
+```
+
+  </TabItem>
+
+  <TabItem value="pages" label="Pages Router">
+
+```ts
+import type { NextApiRequest, NextApiResponse } from 'next';
+import crypto from 'node:crypto';
+
+function utcDateString(ms: number): string {
+	return new Date(ms)
+		.toISOString()
+		.replace(/-/g, '/')
+		.replace(/T/, ' ')
+		.replace(/\.\d+Z$/, '+00:00');
+}
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+	if (req.method !== 'POST') {
+		return res.status(405).json({ error: 'Method Not Allowed' });
+	}
+
+	// expire 1 hour from now (this must be milliseconds)
+	const expires = utcDateString(Date.now() + 1 * 60 * 60 * 1000);
+	const authKey = process.env.TRANSLOADIT_KEY;
+	const authSecret = process.env.TRANSLOADIT_SECRET;
+	const templateId = process.env.TRANSLOADIT_TEMPLATE_ID;
+
+	if (!authKey || !authSecret || !templateId) {
+		return res.status(500).json({ error: 'Missing Transloadit credentials' });
+	}
+
+	const params = JSON.stringify({
+		auth: {
+			key: authKey,
+			expires,
+		},
+		template_id: templateId,
+		fields: {
+			// You can use this in your template.
+			userId: req.body.userId,
+		},
+		// your other params like notify_url, etc.
+	});
+
+	const signatureBytes = crypto
+		.createHmac('sha384', authSecret)
+		.update(Buffer.from(params, 'utf-8'));
+	// The final signature needs the hash name in front, so
+	// the hashing algorithm can be updated in a backwards-compatible
+	// way when old algorithms become insecure.
+	const signature = `sha384:${signatureBytes.digest('hex')}`;
+
+	res.status(200).json({ expires, signature, params });
+}
+```
+
+  </TabItem>
+</Tabs>
+
+On the client we want to fetch the signature and params from the server. You may
+want to send values from React state along to your endpoint, for instance to add
+[`fields`](https://transloadit.com/docs/topics/assembly-variables/) which you
+can use in your template as global variables.
+
+```js
+// ...
+function createUppy() {
+	const uppy = new Uppy();
+	uppy.use(Transloadit, {
+		async assemblyOptions() {
+			// You can send meta data along for use in your template.
+			// https://transloadit.com/docs/topics/assembly-instructions/#form-fields-in-instructions
+			const { meta } = uppy.getState();
+			const body = JSON.stringify({ userId: meta.userId });
```

Comment:
I think userId is not an ideal example from a security perspective. It's better to get the `userId` from the session on the server side than a post payload for tampering reasons. How about... the example becomes about avatar uploading, and a post field we accept is `autoCropFace` or sth?